src/modules/general/slash-commands/set-profile.command.ts

```diff
@@ -0,0 +1,170 @@
+          .setEmoji("✅")
+          .setLabel("Выбрать текущий пресет")
+      );
+
+    const paginationHelper = new PaginationHelper(presets, {
+      elementsOnPage: 1,
+    });
+
+    const [firstPreset] = paginationHelper.createPage();
+
+    const componentCollector =
+      interaction.channel?.createMessageComponentCollector({
+        componentType: ComponentType.Button,
+        filter: (int) =>
+          int.user.id === interaction.user.id &&
+          int.message.interaction?.id === interaction.id,
+        time: 5 * 1000 * 60000, // 5 minutes
+      });
+
+    if (!componentCollector) {
+      return this.sendError("Can't create component collector :(", interaction);
+    }
+
+    const { tokens } = gatherProfileTokens(
+      userData,
+      interaction.member as GuildMember,
+      this.client.voiceManager
+    );
+
+    await interaction.editReply({
+      content: this.createContent(paginationHelper),
+      embeds: [this.createEmbedFromJson(tokens, firstPreset)],
+      components: [actionRow, actionRowProceed],
+    });
+
+    return componentCollector.on("collect", async (int) => {
+      let preset: ProfilePresetEntity | undefined;
+
+      switch (int.customId) {
+        case NavigationButtons.TO_LEFT:
+          preset = paginationHelper.prevPage().createPage()[0];
+          break;
+        case NavigationButtons.TO_RIGHT:
+          preset = paginationHelper.nextPage().createPage()[0];
+          break;
+        case NavigationButtons.TO_STOP:
+          componentCollector.stop();
+          interaction.deleteReply();
+          return;
+        case TO_PROCEED:
+          preset = paginationHelper.currentPage;
+          userData.selected_preset = preset.id !== "default" ? [preset] : []; // If default preset, then remove it
+
+          await UserEntity.save(userData);
+          this.sendSuccess("Ваш профиль успешно обновлен! 🎉", interaction);
+          return;
+      }
+
+      if (!preset) {
+        return;
+      }
+
+      int.update({
+        content: this.createContent(paginationHelper),
+        embeds: [this.createEmbedFromJson(tokens, preset)],
+      });
+    });
+  }
+
+  public createEmbedFromJson(
+    tokens: { [k: string]: string | number },
+    preset: ProfilePresetEntity
+  ) {
+    const embed = preset.json || JSON.stringify(defaultTemplate);
+    const embedBuilder = embed.replace(/{{(.*?)}}/g, (_, mtch) => {
+      return `${tokens[mtch].toString().replace(/"/g, '\\"')}`;
+    });
+
+    return new EmbedBuilder(JSON.parse(embedBuilder));
+  }
+
+  public createContent(
+    paginationHelper: PaginationHelper<ProfilePresetEntity>
+  ) {
+    return (
+      `### 📃 Текущий пресет: [${paginationHelper.page} | ${paginationHelper.totalPages}]\n` +
+      `Используйте стрелки ниже чтобы выбрать подходящий пресет\n` +
+      `Нажмите кнопку с ✅ чтобы выбрать текущий пресет\n` +
+      `Нажмите кнопку с ❌ чтобы закрыть меню\n` +
+      `### Превью пресета:`
+    );
+  }
```

Comment:
we have same methods in another command (profile-presets) would be great to move that functionality to util/helper function and reuse it. e.g. remove this methods from both commands and create two functions for them and reuse where needed

Reply:
I admire the fact that we want to reuse the same code, but this is not the case here. While it can be argued for `createEmbedFromJson` function, not the case for `createContent` at all. It returns completely different text. 
But I will export both variant of `createEmbedFromJson` function (btw this resolve [this](https://trello.com/c/ZCmAJOWs/20-parse-tokens-on-profile-preset-list) bug) into one.